### ISSUE TITLE
Add a Tips & Tricks Section to Godot Docs

### ISSUE THREAD

**Issue by mrjustaguy** [godotengine/godot-proposals#3625]
### Describe the project you are working on

N/A - Just Experimenting with Godot and it's capabilites

### Describe the problem or limitation you are having in your project

The Docs mostly revolve around fairly basic explanations of how things work, but don't often offer up too many tips on how to actually best use a specific feature.

This is especially clear on the Rendering side of things, as this is the most complex and yet the most clearly visible part of Games that also requires a lot of artist work, where the theory around how specific things work, doesn't really help artists too much with using those tools to get good results if they don't have much experience to start with.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Creating a Database of Tips and Tricks to get certain common effects and Having it available in the official Docs would help Users create even better content, faster, removing the need for delving too deep into the engine for people who aren't ready for that and don't desire to do something too exotic.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Add a Tips & Tricks Section and Split it into several general categories such as:
1) Rendering
2) Coding
3) Physics
4) Misc
(More could be added, depending really on how much interest there is from the community in finding and documenting things that fall into this useful category that doesn't fit into these sub-categories)

Here are a few that I'd include that I believe would be useful for people:
1) (RENDERING) Quick & cheap believable Reflections - Add a Reflection Probe as a Child of the Player Camera and set refreshing to Always.
Pros: Easy to setup, fully dynamic (Doesn't require placing probes around The World) 
Cons: Properly manually placed Reflection Probes provide superior quality Reflections, Objects between Camera and a Reflective surface will in some scenarios have clearly inaccurate Reflections
2) (CODING) Easy Save System - when you want to save, use Propagate call From Root Node to turn every object's owner to the node that Propagated the call. Any Variables in Scripts that you want saved should use export to be saved. Pack Root Node with Packed Scene, after that Save the Packed Scene into a .tscn/.scn and to load, just use change scene to the desired save scene.
Pros: Implementation needs only half a dozen lines of GDScript
Cons: Every Variable that is to be stored from scripts has to be Exported in order to be saved with the scene, and more efficient save systems can be implemented by skilled programmers that work faster and/or better use space.

These are Two I could think of off the top of my head, If you know of any useful tips & tricks that you think can help others deal with common game dev issues (that aren't caused by a bug in Godot or a feature that should be implemented in Godot but isn't) please comment them below

### If this enhancement will not be used often, can it be worked around with a few lines of script?

N/A

### Is there a reason why this should be core and not an add-on in the asset library?

It's Documentation, should be in Godot's Documentation, instead of spread out around the internet, for the more useful/common tips & tricks.

**Comment by pycbouh** [comment]
That type of content doesn't fit the idea that we have for the docs. We do indeed want the docs to provide dry-ish formal representation of how to use the engine and how to get started. Tips and tricks are better serviced by third-parties and content creators. Like, recipes by KidsCanCode.

**Comment by Calinou** [comment]
See also https://github.com/godotengine/godot-proposals/issues/15.

**Comment by Cerno-b** [comment]
My 2 cents on this:

What if it's more of a collection of easily overlooked Godot features or shortcuts? Sometimes a feature is added to Godot just to make things easier (similar to syntactic sugar in coding). Or there might be some known bad habits that have easy to explain better solutions. Having a collection of very compressed, easy to digest nuggets of truth that create a "wow, I didn't know this exists" experience might be helpful.

TBH, I came here from https://github.com/godotengine/godot-proposals/issues/15 and kind of think that having a collection of these bite-sized info blocks would be a good first step in that direction (although the proposal has been rejected). If this was ever attempted again, having a collection of data to feed into a system like that would be nice to have. Also grouping the tips under headings to give them some structure could pave the way to have more curated TOTDs in Godot (like "don't show me 3D tips").

Real world example: I was using the German version of Godot while testing the translations and realized that all the properties in the inspector were in English. I was made aware in the chat that there is a setting for changing them to translated, which I wouldn't have found on my own. Since then I encountered another German who didn't know he could change the property language (and apparently never bothered to look for a way until I mentioned the option).

**Comment by Cerno-b** [comment]
@AThousandShips Yes, that would be more like a "recipe" section in a forum or a blog post. What I had in mind was really more of a concept that can be explained in a few lines of text.

![image](https://github.com/godotengine/godot-proposals/assets/16470072/9bb56c92-fd1f-4e41-be62-dd3a26311861)

Also something like: "A frequent beginner mistake is X, consider doing Y instead", but not in the form of a tutorial, but rather a quick reminder that what I am doing could be done better. Instead of a lengthy description of how to do it, there could be a link to the docs (e.g. to the best practices section).

**Comment by Cerno-b** [comment]
Something like this I just stumbled upon in the manual: 

> If you run into performance issues when using 2D lights, it may be worth replacing some of them with Sprite2D nodes that use additive blending

Maybe with small before/after image to illustrate. The point is to give a peek into a cool little trick, portrayed as conventiently as possible so that the reader can identify at a glance if this could be helpful or not.

**Comment by tetrapod00** [comment]
The scope of what we can keep and maintain in the official docs is fairly narrow, as already mentioned.

But it's worth noting that since [comments/user contributed notes](https://github.com/godotengine/godot-docs-user-notes/discussions) were added, if you feel like you have a tip or trick worth sharing, you can post it yourself as a comment on a relevant page. The barrier to entry to doing so is a lot lower than going through a PR review process to add something to the official docs, and it will still be visible to readers of the page.
